ambari/datadog_checks/ambari/ambari.py

```diff
@@ -0,0 +1,176 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from six import iteritems
+import requests
+
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.errors import CheckException
+from . import common
+
+# Tag templates
+CLUSTER_TAG_TEMPLATE = "ambari_cluster:{}"
+HOST_TAG = "ambari_host:"
+SERVICE_TAG = "ambari_service:"
+COMPONENT_TAG = "ambari_component:"
+
+# URL queries
+COMPONENT_METRICS_QUERY = "/components?fields=metrics"
+SERVICE_INFO_QUERY = "?fields=ServiceInfo"
+
+# Response fields
+METRICS_FIELD = "metrics"
+
+
+class AmbariCheck(AgentCheck):
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        super(AmbariCheck, self).__init__(name, init_config, agentConfig, instances)
+        self.hosts = []
+        self.clusters = []
+        self.services = []
+
+    def check(self, instance):
+        server = instance.get("url", "")
+        port = str(instance.get("port", ""))
+        path = str(instance.get("path", ""))
+        base_tags = instance.get("tags", [])
+        services = instance.get("services", [])
+        metric_headers = [str(h) for h in instance.get("metric_headers", [])]
+
+        clusters = self.get_clusters(server, port)
+        self.get_host_metrics(clusters, server, port, base_tags)
```

Comment:
I had added some logic to check the paramaters `collect_host_metrics` and `collect_service_metrics`, but for some reason I removed it.  This would basically be a way for users to configure if they wanted to limit collection of those types of metrics.  See [here](https://github.com/DataDog/integrations-core/commit/e3ec7b66d6915082d721204ea522e8b06f77826d#diff-2d767211cfd53ee87137307b8d7334daL30) for details - would probably be worth adding back in.

Reply:
Thanks for the tip. I've also added colllect_service_status